src/lib.rs

```diff
@@ -95,9 +112,43 @@ impl Decoder {
             if let Some(duration) = duration {
                 data.m.duration = duration;
             }
-            let ret = dav1d_send_data(self.dec, &mut data);
+            let ret = dav1d_send_data(self.dec.as_ptr(), &mut data);
             if ret < 0 {
-                Err(Error(ret))
+                let ret = Error(ret);
+
+                if ret.is_again() {
+                    self.pending_data = Some(data);
+                } else {
+                    dav1d_data_unref(&mut data);
+                }
+
+                Err(ret)
+            } else {
+                Ok(())
+            }
+        }
+    }
+
+    pub fn send_pending_data(&mut self) -> Result<(), Error> {
+        let mut data = match self.pending_data.take() {
+            None => {
+                return Ok(());
```

Comment:
Might want to make this a panic or a `EAGAIN`, not sure. (`get_picture()` returns `EAGAIN` if more data is needed, which is the case here too).

What do you think?

Reply:
`Ok(())` seems fine here, if there is no more pending data you can resume feeding in data.